### PR TITLE
Document release procedure.

### PR DESCRIPTION
**Pull request by marun**

### CODE REVIEW

File: docs/releasing.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+### Releasing Federation v2
+
+Creating a federation v2 release involves the following steps:
+
+ - Localling create an annotated tag in the format v[0-9]+.[0-9]+.[0-9]+
```
Comment (irfanurrehman) [inline docs/releasing.md]: s/Localling/Locally

File: docs/releasing.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+   - `git tag -a <tag> -m "Creating release tag <tag>"`
+   - An annotated tag is required to trigger the image build and to
+     ensure that the `kubefed2` binary is versioned correctly.
```
Comment (font) [inline docs/releasing.md]: Also the `controller-manager` binary.

File: docs/releasing.md (hunk 1 above)
Comment (font) [inline docs/releasing.md]: How about an ordered list?

File: docs/releasing.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+   - At the time of writing it is not possible to create an annotated
+     tag through the github web interface.
+ - Push the tag to master
+   - `git push origin <tag>`
+   - This requires write access to the repo.
+ - Build `kubefed2` for release
+   - `make kubefed2`
+   - `cd bin`
```
Comment (gyliu513) [inline docs/releasing.md]: How about adding a command as follows to confirm the `kubefed2` version?

```shell
kubefed2 version
```

Comment (irfanurrehman) [reply docs/releasing.md]: both `kubefed2` and `controller-manager` have a version command option.

File: docs/releasing.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+### Releasing Federation v2
+
+Creating a federation v2 release involves the following steps:
+
+ - Localling create an annotated tag in the format v[0-9]+.[0-9]+.[0-9]+
+   - `git tag -a <tag> -m "Creating release tag <tag>"`
+   - An annotated tag is required to trigger the image build and to
```
[the remark this answers is not in the document]
Comment (marun) [reply docs/releasing.md]: I think I conflated the image build with the binary build.  I've removed mention of the image build requiring an annotated tag.

File: docs/releasing.md [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+   - `git push origin <tag>`
+   - This requires write access to the repo.
+ - Build `kubefed2` for release
+   - `make kubefed2`
+   - `cd bin`
+   - `tar cvzf kubefed2.tar.gz kubefed2`
+   - `sha256sum kubefed2.tar.gz > kubefed2.tar.gz.sha`
+ - Create github release
+   - Copy text from old release and replace old tag references
+   - Add `kubefed2.tar.gz` and `kubefed2.tar.gz.sha`
```
[the remark this answers is not in the document]
Comment (marun) [reply docs/releasing.md]: Actually the binaries can be added both on creation of the release and at any time after creation.

**Pull request merged**